shm/Map.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "Island.hpp"
+#include <vector>
+
+class Map {
+private:
+    Island* currentPosition_ {};
```

Comment:
Do we need {} for this pointer initialization?

Reply:
So I've initially thought that empty initializer list would initialize the pointer with nullptr. Turns out it's not the case. It is initialized by some null value. I've changed to reflect what I wanted it to do.